api/callbacks.go

```diff
@@ -103,8 +123,9 @@ var db_vtable = C.DB_vtable{
 
 // contract: original pointer/struct referenced must live longer than C.DB struct
 // since this is only used internally, we can verify the code that this is the case
-func buildDB(kv KVStore) C.DB {
+func buildDB(kv KVStore, gm GasMeter) C.DB {
```

Comment:
yeah, so I don't mind doing it the way you described, but the pointer issue doesn't come from this additional level of indirection - by default Go types are value-types (unlike python, java, etc), and GasMeter contains no internal pointer. On the other hand, i am concerned about the KVStore that we already pass around between all the layers, because it contains a bunch of internal pointers. But for some reason the dynamic pointer validation tool didn't report any problem with it even on the strictest setting, and we didn't observe any issue with it in practice. Sooo... This should be ok?

Reply:
Let's try that (on a branch) and if it works with struct settings, great. If not, no good.

Notice that we don't hold those Go pointers in C/Rust memory after the function call returns, so that remove the big issue about unbounded behavior (external references to garbage collected items)

Reply:
note for myself:
the `gm GasMeter` field should be `gm *GasMeter` in order to stay in sync with the one the SDK manages.

Reply:
Two notes:
* GasMeter is an interface type, which already means that it's a pointer and vtable, not a value type.
* https://github.com/CosmWasm/go-cosmwasm/pull/92 shows that this approach is not acceptable by cGo safety rules :frowning_face: